### PR TITLE
Add missing primaryConstructor & companionObject to TypeSpec#toBuilder()

### PR DESCRIPTION
**Pull request by hzsweers**

### CODE REVIEW

**Comment by hzsweers** [comment]
Not sure how you'd want to test this. If something like `typeSpec.toBuilder().build() == typeSpec` is supposed to hold true, I could make a comprehensive smoke test that covers all the bases to make sure they're still there

**Comment by Egorand** [comment]
Yes, that'd be great. `equals()` uses `toString()`, which just emits into a `StringBuilder`, hence it should work for tests.

**Comment by hzsweers** [comment]
Done! 3f77f64

File: src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt [hunk 1]
```diff
@@ -19,11 +19,15 @@ import com.google.common.collect.ImmutableMap
...
 import com.squareup.kotlinpoet.KModifier.ABSTRACT
+import com.squareup.kotlinpoet.KModifier.DATA
+import com.squareup.kotlinpoet.KModifier.IN
+import com.squareup.kotlinpoet.KModifier.INNER
 import com.squareup.kotlinpoet.KModifier.INTERNAL
 import com.squareup.kotlinpoet.KModifier.PRIVATE
 import com.squareup.kotlinpoet.KModifier.PUBLIC
 import com.squareup.kotlinpoet.KModifier.VARARG
 import com.squareup.kotlinpoet.jvm.throws
+import org.eclipse.jdt.internal.compiler.parser.Parser.name
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt]: Is this used?

Comment (hzsweers) [reply src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt]: 👀  I have no idea where that came from. Will remove

Comment (hzsweers) [reply src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt]: d01abec

File: src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt [hunk 2]
```diff
@@ -2140,6 +2143,40 @@ class TypeSpecTest {
...
+        .superclass(ClassName("texmexfood", "TortillaBased"))
+        .addSuperclassConstructorParameter("true")
+        .addProperty(PropertySpec.builder("meat", ClassName("texmexfood", "Meat"))
+            .build())
+        .addFunction(FunSpec.builder("fold")
+            .build())
+        .addSuperinterface(ClassName("texmexfood", "Consumable"))
+        .build()
+
+    assertThat(comprehensiveTaco.toBuilder().build()).isEqualTo(comprehensiveTaco)
```
Comment (swankjesse) [inline src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt]: Nice test.

**Pull request merged**